### PR TITLE
charts: allow to set deployment update strategy

### PR DESCRIPTION
**Pull request by bclermont**
## What is the problem I am trying to address?

If [`storage.disk.persistence.enabled`](https://github.com/gomods/athens/blob/6b5b62c0fceb97e8c86816fcf82a93f730a330e3/charts/athens-proxy/values.yaml#L49) is set to `true` and the `accessMode` is not `ReadWriteMany`. the default strategy must be switched to `type: Recreate`. if not it will fail as the other pod is using the PVC.

### CODE REVIEW

File: charts/athens-proxy/templates/deployment.yaml [hunk 1]
```diff
@@ -9,6 +9,13 @@ metadata:
     heritage: "{{ .Release.Service }}"
 spec:
   replicas: {{ .Values.replicaCount }}
+{{- if .Values.strategy }}
+  strategy:
+{{ toYaml .Values.strategy | indent 4 }}
+  {{- if eq .Values.strategy.type "Recreate" }}
```
Comment (twexler) [inline charts/athens-proxy/templates/deployment.yaml]: This clause seems odd. I assume it's a bit of defensive programming, but if a user is changing the deployment strategy they should probably not be specifying values that aren't compatible.

Comment (bclermont) [reply charts/athens-proxy/templates/deployment.yaml]: @twexler 

[this is the reason why I did that](https://github.com/helm/charts/commit/663bd1f7484877f4fcfa7ae9a9139cb198d7e659)

if you're still uncomfortable with that, I can remove it

Comment (twexler) [reply charts/athens-proxy/templates/deployment.yaml]: Ah, okay. This is a common pattern, interesting.

**Comment by twexler** [comment]
@bclermont I'm going to wait to merge this until we merge #1595 (as it also updates this chart). We're working through some issues with Netlify right now and want to make sure those are solved before that merges.

**Comment by arschles** [comment]
@twexler @bclermont I've merged #1595, so after conflicts are resolved this is ok to go. The conflicts will just be to update the `version` field in the chart.yaml

**Comment by bclermont** [comment]
@twexler done

**Pull request merged**